super-test.sh

```diff
@@ -403,13 +403,25 @@ if [ $IS_CLANG = yes ]; then
   # Don't fail out on this ridiculous "argument unused during compilation" warning.
   export CXXFLAGS="$CXXFLAGS -Wno-error=unused-command-line-argument"
 
-  # At the moment, only our clang-10 CI run seems to like -fcoroutines-ts. Earlier versions seem to
-  # have a library misconfiguration causing ./configure to result in the following error:
-  #   conftest.cpp:12:12: fatal error: 'initializer_list' file not found
-  #   #include <initializer_list>
-  # Let's use any clang version >= 10 so that if we move to a newer version, we'll get additional
-  # coverage by default.
-  if [ "${CXX#*-}" -ge 10 ] 2>/dev/null; then
+  # Enable coroutines if supported.
+  if [ "${CXX#*-}" -ge 14 ] 2>/dev/null; then
+    # Somewhere between version 10 and 14, Clang started supporting coroutines as a C++20 feature,
+    # and started issuing deprecating warnings for -fcoroutines-ts. (I'm not sure which version it
+    # was exactly since our CI jumped from 10 to 14, so I'm somewhat arbitrarily choosing 14 as the
+    # cutoff.)
+    export CXXFLAGS="$CXXFLAGS -std=c++20 -stdlib=libc++"
```

Comment:
I wonder if we lose any code coverage by using `-std=c++20` instead of `gnu++20`? Maybe we don't actually use any GNU extensions that clang doesn't provide by default.

Reply:
I would expect that if we were using any "GNU extensions" that aren't enabled for `-std=c++20`, we'd see errors, because I'm not aware of us using #ifdefs that somehow distinguish whether GNU extensions are enabled. So since we see no errors, I suspect it doesn't really make a difference?